sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSession.java

```diff
@@ -583,16 +590,19 @@ private LinkSubscription<AmqpReceiveLink> getSubscription(String linkName, Strin
 
         receiver.setSource(source);
 
-        final Target target = new Target();
-        receiver.setTarget(target);
-
         // Use explicit settlement via dispositions (not pre-settled)
         receiver.setSenderSettleMode(senderSettleMode);
         receiver.setReceiverSettleMode(receiverSettleMode);
 
+        final Target target = new Target();
         if (receiverProperties != null && !receiverProperties.isEmpty()) {
             receiver.setProperties(receiverProperties);
+            final String clientIdentifier = (String) receiverProperties.get(CLIENT_RECEIVER_IDENTIFIER);
```

Comment:
Is the property only settable by owner client, or it can be set by a different sender and possible set a non-String object to it?
It would need protection, if latter.

Reply:
Do you mean it could be passed a customized property with non-String in this field? In my gut feeling, the clients don't expose the method of creating link or session, so the latter won't happen. I'll check this logic.

Reply:
Sure. Do note that this code is in azure-core-amqp. Likely it is only eh and sb uses it, but theoretically it is still a stand-alone lib that can be used by anybody. Think it on the API of this lib.

If any usage would cause it crash here, we generally have 2 choices: crash with a meaningful exception (and fail-fast), or no crash (latter seems easier).